### PR TITLE
Attachment support

### PR DESCRIPTION
**Pull request by hdoddema**
Hello,

I have implemented sending attachments and inline attachments according to [the Mailjet API v3 documentation](https://dev.mailjet.com/guides/#sending-with-attached-files-v3).

`attach`, `attachContent`, `embed` and `embedContent` should be available now.

### CODE REVIEW

File: Mailer.php [hunk 1]
```diff
@@ -131,30 +131,55 @@ public function getResponse()
...
         $body = [
             'Subject' => $message->subject,
             'Text-part' => $message->textBody,
             'Html-part' => $message->htmlBody,
-            'Recipients' => $recipients,
+            'To' => join(', ',$to),
         ];
 
+        if ($cc) {
+            $body['Cc'] = join(', ',$cc);
```
Comment (Julian-B90) [inline Mailer.php:170]: missing space `join(', ', $cc);`

File: Mailer.php [hunk 2]
```diff
@@ -131,30 +131,55 @@ public function getResponse()
...
+        }
```
Comment (Julian-B90) [inline Mailer.php:171]: Add Space

File: Message.php [hunk 3]
```diff
@@ -113,6 +117,9 @@ public function getCc() {
      * @inheritdoc
      */
     public function setCc($cc) {
+        if (!is_array($cc)){
```
Comment (Julian-B90) [inline Message.php]: Please add space `if (!is_array($cc)) {`

File: Message.php [hunk 4]
```diff
@@ -128,6 +135,9 @@ public function getBcc() {
      * @inheritdoc
      */
     public function setBcc($bcc) {
+        if (!is_array($bcc)){
```
Comment (Julian-B90) [inline Message.php:138]: Please add space `if (!is_array($bcc)) {`

**Review by Julian-B90 — COMMENTED** [review]
Can you maybe add an example for Attachments and InlinedAttachments

**Comment by hdoddema** [comment]
Oh yes, and in the meantime I have also changed Recipients to separate fields for To, CC and BCC. It seems to work ok, but I'm not sure if using Recipients had any advantages over doing it like this. 

But this way it also supports setBcc and setCc as you would expect.

I have fixed a few bugs with the inline attachments just now.

Here is some sample code.

```php
// Mail with attachment from string via Message::attachContent()
Yii::$app->mailer->compose('view-name')
->setSubject('Mail with attachment from content')
->attachContent("This is the attachment content",['fileName'=>'attachment.txt', 'contentType'=>'text/plain'])
->setTo('info@example.com')
->send();

// Mail with attachment from file via Message::attach()
$filePath = ... // a file path here;
Yii::$app->mailer->compose('view-name')
->setSubject('Mail with attachment from content')
->attach($filePath)
->setTo('info@example.com')
->send();
```
Embedding allows you to use the image directly in the HTML content. It does require a different approach because you need a reference to the message object to get the image links.


```php
$message = Yii::$app->mailer->compose('view-name')
    ->setSubject('Mail with inline attachment')
    ->setTo('info@example.com')
$imageLink = $message->embed($filePath);
$message->setHtmlBody('Sample embedded image <br> <img src="'.$imageLink.'">');
$message->send();
```

**Review by Julian-B90 — APPROVED** [review]

**Comment by Julian-B90** [comment]
Thank you very much for your work 👍

**Pull request merged**